core/domain/exp_jobs_one_off_test.py

```diff
@@ -2102,3 +2102,47 @@ def test_no_action_is_performed_for_deleted_exploration(self):
 
         run_job_for_deleted_exp(
             self, exp_jobs_one_off.ExpSummariesContributorsOneOffJob)
+
+
+class TestDeleteStateIdMappingModelsOneOffJobs(test_utils.GenericTestBase):
+    """Tests the state ID mapping deletion job."""
+    def test_job_deletes_all_instances_of_model(self):
+        exp_models.StateIdMappingModel.create(
+            'exp_id', 1, {'state_1': 1}, 1)
+        exp_models.StateIdMappingModel.create(
+            'exp_id', 2, {'state_1': 1}, 1)
+        exp_models.StateIdMappingModel.create(
+            'exp_id', 3, {'state_1': 1}, 1)
+
+        self.assertEqual(self.count_jobs_in_taskqueue(
+            taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 0)
+
+        job_id = (
+            exp_jobs_one_off.DeleteStateIdMappingModelsOneOffJobs.create_new())
+        exp_jobs_one_off.DeleteStateIdMappingModelsOneOffJobs.enqueue(job_id)
+
+        self.assertEqual(self.count_jobs_in_taskqueue(
+            taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
+
+        self.assertEqual(self.count_jobs_in_taskqueue(
+            taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 0)
+
+        with self.assertRaisesRegexp(
+            base_models.BaseModel.EntityNotFoundError,
+            'Entity for class StateIdMappingModel with id exp_id.1 not found'):
+            exp_models.StateIdMappingModel.get_state_id_mapping_model(
+                'exp_id', 1)
+
+
+        with self.assertRaisesRegexp(
+            base_models.BaseModel.EntityNotFoundError,
+            'Entity for class StateIdMappingModel with id exp_id.2 not found'):
+            exp_models.StateIdMappingModel.get_state_id_mapping_model(
+                'exp_id', 2)
+
+        with self.assertRaisesRegexp(
+            base_models.BaseModel.EntityNotFoundError,
+            'Entity for class StateIdMappingModel with id exp_id.3 not found'):
+            exp_models.StateIdMappingModel.get_state_id_mapping_model(
```

Comment:
Is there any way to test that the error isn't for models just marked as deleted? Also, do we have to mark strict=True to get the error?

Reply:
I made it so that we test existence before and non-existence after the job. I think that should be a good test? 

Regarding strict = True, it works without it too.

Reply:
The endline verification says it's either deleted or marked as deleted (deleted=True).
https://github.com/oppia/oppia/blob/a569a29fd4c938274333fa01d66b188636a0b904/core/storage/base_model/gae_models.py#L95-L103

But I don't think we need to test that as well! (I'm going to merge the PR)

> Regarding strict = True, it works without it too.

I saw the default value of `strict` is True. Thanks for checking!